core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -455,8 +455,21 @@ private void doStop(CompletableFuture<Void> future,
         }
 
         private void finishDoStop(CompletableFuture<Void> future) {
-            // TODO(trustin): Add shutdownBlockingTaskExecutorOnStop
-            // TODO(trustin): Count the pending blocking tasks and wait until it goes zero.
+            if (config.shutdownBlockingTaskExecutorOnStop()) {
+                if (config.blockingTaskExecutor() instanceof InterminableExecutorService) {
+                    final ExecutorService service =
+                            ((InterminableExecutorService) config.blockingTaskExecutor()).getExecutorService();
+                    try {
+                        service.shutdown();
+                        while (!service.isTerminated()) {
+                            service.awaitTermination(500, TimeUnit.MILLISECONDS);
+                        }
+                    } catch (Exception e) {
+                        logger.warn("Failed to shutdown the {}:", service.getClass().getSimpleName(), e);
```

Comment:
For anonymous class, `getSimpleName()` will not give useful information. How about just passing `service`?